rsocket-core/src/main/java/io/rsocket/core/CleanOnClearQueueDecorator.java

```diff
@@ -0,0 +1,108 @@
+package io.rsocket.core;
+
+import io.netty.util.ReferenceCountUtil;
+import io.rsocket.Payload;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Queue;
+
+final class CleanOnClearQueueDecorator implements Queue<Payload> {
```

Comment:
This is a temporary workaround for https://github.com/reactor/reactor-core/pull/2114, right? Maybe insert a brief implementation comment to that extent and with a reference to the issue.

Reply:
Yes, it makes sense. In any case, this is package-private so nobody should use it